horizontal_rule_test.py

```diff
@@ -1,6 +1,54 @@
+import pytest
 from horizontal_rule import HorizontalRule
 
 
 def test_render():
     horiz_rule = HorizontalRule('')
     assert horiz_rule.render() == '<hr>'
+
+
+@pytest.mark.parametrize(
+    "content, start, end",
+    [
+        ('***', 0, 3),
+        ('____', 0, 4),
+        ('-----', 0, 5),
+        (
+            '''---
+            abc
+            ''', 0, 3
+        ),
+        (
+            '''abc
+*****
```

Comment:
I am consider this one, the "*****" is a new paragraph right?

Reply:
not sure what you mean?

Reply:
I mean, in this case, you have 3 paragraphs, first one is "abc", second is "*****", third is "        ". So the start begin should be "0, 5" which indicates the relative location inside that paragraph. in other words, cross pargraph parsing should not be supported.

Reply:
Feel like you treat single line as a Paragraph? then how do you support the fenced code block like this

```python
print('hello Andy')
print('hello VC')
```

Reply:
the different is, for Text/FenceBlock/Quote block, they support cross line paragraph. 
For header/horizontal line, they don't do crosse line parse.